packages/common/src/injectDecoratorFactory.ts

```diff
@@ -0,0 +1,14 @@
+import { Inject } from '@nestjs/common';
+
+/**
+ * Creates a decorator that can be used as a convenience to inject a specific token
+ *
+ * Instead of using @Inject(SOME_THING_TOKEN) this can be used to create a new named Decorator
+ * such as @InjectSomeThing() which will hide the token details from users making APIs easier
+ * to consume
+ * @param token
+ */
+export const makeInjectableDecorator = (
+  token: string | symbol
+): (() => ParameterDecorator) => () => (target, key, descriptor) =>
+  Inject(token)(target, key, descriptor);
```

Comment:
Instead of having to say `Inject(token)(target, key, descriptor)`, couldn't you just return the `Inject(token)` and let typescript take care of the rest? Overall making the function
```ts
export const makeInjectableDecorator = (token: string | symbol) => Inject(token);
```

Reply:
I don't think so. That would return the result of calling Inject instead of a function that calls Inject on the thing that it's applied to

Reply:
Oh, you know what, you're right. I was thinking in the context of actually having `makeInjectableDecorator` the decorator, which absolutely doesn't make sense. 

```ts
export function OgmaLogger(topic: string | Function) {
  return Inject(
    createProviderToken(typeof topic === 'function' ? topic.name : topic),
  );
}
```
This is the example I was thinking of, but that's not quite a dynamic use case

Reply:
Technically it could be `export const makeInjectableDecorator = (token: string | symbol) => () =>
  Inject(token);` instead (note the extra arrow) but then the typing no longer aligns with `ParameterDecorator`